core/persist/PersistenceProvider.ts

```diff
@@ -98,29 +122,10 @@ export abstract class PersistenceProvider<S> {
                 if (rest.getData || rest.setData) type = 'custom';
             }
 
-            switch (type) {
-                case 'pref':
-                    ret = new PrefProvider(cfg);
-                    break;
-                case 'localStorage':
-                    ret = new LocalStorageProvider(cfg);
-                    break;
-                case 'sessionStorage':
-                    ret = new SessionStorageProvider(cfg);
-                    break;
-                case `dashView`:
-                    ret = new DashViewProvider(cfg);
-                    break;
-                case `viewManager`:
-                    ret = new ViewManagerProvider(cfg);
-                    break;
-                case 'custom':
-                    ret = new CustomProvider(cfg);
-                    break;
-                default:
-                    throw XH.exception(`Unknown Persistence Provider for type: ${type}`);
-            }
+            const clazz = this.PROVIDERS[type] as any;
+            throwIf(!clazz, `Unknown Persistence Provider type: ${type}`);
 
+            ret = new clazz(cfg) as PersistenceProvider<S>;
```

Comment:
I *think* if the provider has to be a class with a constructor of this form, we might have a way to incorportate that into the type

Reply:
Making a PR now with that type tweak